### PR TITLE
Update pin/unpin icon after pinning notebook

### LINKED ISSUE #19044: Notebook view pin/unpin doesn't update immediately
Not a regression, present in 1.35.1 as well

1. Open up workspace with Notebooks in it
2. Click the "pin" icon next to a Notebook

Nothing appears to happen. It does seem like the pin/unpin action is working in the background though - if I reload the window then the icon displays correctly. The icon just isn't updating correctly.

### PR DESCRIPTION
**Pull request by barbaravaldez**
**UPDATE: I reverted my changes on the getSections method. We should create a separate item to fix the current book logic, because we use it in other functionalities as well.** 

Update icon after pinning/unpinning notebook immediately after firing action. Currently, we create a new copy of the  bookTreeItem object when adding a pinned notebook to the Pinned Notebook Tree View, and then we just close the notebook in the Pinned Notebooks Tree View, which doesn't update the tree item in the Book Tree View. 

We need to look for the tree item that was pinned  in the Book Tree View and fire the tree data change event on it. 

This PR fixes  #19044

https://user-images.githubusercontent.com/34872381/183587654-5caf5006-c561-4d11-9e6f-d0bb03c48035.mov

### CODE REVIEW

File: extensions/notebook/src/common/utils.ts [hunk 1]
```diff
@@ -445,6 +445,7 @@ export function isBookItemPinned(notebookPath: string): boolean {
 export enum BookTreeItemType {
 	Book = 'Book',
 	Notebook = 'Notebook',
+	BookNotebook = 'BookNotebook',
```
Comment (chlafreniere) [inline extensions/notebook/src/common/utils.ts:448]: Why do we need another item type here?

File: extensions/notebook/src/book/bookPinManager.ts [hunk 2]
```diff
@@ -58,7 +58,8 @@ export class BookPinManager implements IBookPinManager {
 			pinnedBooks.splice(existingBookIndex, 1);
 			modifiedPinnedBooks = true;
 		} else if (existingBookIndex === -1 && operation === PinBookOperation.Pin) {
-			let addNotebook: IPinnedNotebook = { notebookPath: bookPathToChange, bookPath: notebook.book.root, title: notebook.book.title };
+			const bookPath = notebook.tableOfContents.sections ? notebook.book.root : undefined;
```
Comment (MaddyDev) [inline extensions/notebook/src/book/bookPinManager.ts:61]: if we're not using this bookPath anywhere else, just add the undefined check inline.

File: extensions/notebook/src/common/utils.ts (hunk 1 above)
Comment (MaddyDev) [reply extensions/notebook/src/common/utils.ts:448]: BookNotebook is very confusing, what are you trying to refer here PinnedNotebook ?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 3]
```diff
@@ -264,7 +270,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	async addNotebookToPinnedView(bookItem: BookTreeItem): Promise<void> {
 		let notebookPath: string = bookItem.book.contentPath;
 		if (notebookPath) {
-			let notebookDetails: IPinnedNotebook = bookItem.book.root ? { bookPath: bookItem.book.root, notebookPath: notebookPath, title: bookItem.book.title } : { notebookPath: notebookPath };
+			let notebookDetails: IPinnedNotebook = bookItem.book.tableOfContents.sections ? { bookPath: bookItem.book.root, notebookPath: notebookPath, title: bookItem.book.title } : { notebookPath: notebookPath };
```
Comment (MaddyDev) [inline extensions/notebook/src/book/bookTreeView.ts:273]: what are you checking for tableOfContents.sections and not using it in the following ternary statement.

File: extensions/notebook/src/common/utils.ts (hunk 1 above)
Comment (barbaravaldez) [reply extensions/notebook/src/common/utils.ts]: I removed this extra type

File: extensions/notebook/src/book/bookTreeView.ts (hunk 3 above)
Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTreeView.ts]: This is to differentiate an standalone notebook from a book notebook. Retrieving the tree item is different in each case.

File: scripts/test-extensions-unit.sh [hunk 4]
```diff
@@ -91,90 +91,90 @@ echo "VSCODEEXTDIR : '$VSCODEEXTDIR'"
...
-echo ************************************************
-echo *** starting machine-learning tests ***
-echo ************************************************
-"$INTEGRATION_TEST_ELECTRON_PATH" $LINUX_EXTRA_ARGS --extensionDevelopmentPath=$ROOT/extensions/machine-learning --extensionTestsPath=$ROOT/extensions/machine-learning/out/test $ALL_PLATFORMS_API_TESTS_EXTRA_ARGS
-
-# echo ******************************************
-# echo *** starting mssql tests ***
-# echo ******************************************
-# "$INTEGRATION_TEST_ELECTRON_PATH" $LINUX_EXTRA_ARGS --extensionDevelopmentPath=$ROOT/extensions/mssql --extensionTestsPath=$ROOT/extensions/mssql/out/test $ALL_PLATFORMS_API_TESTS_EXTRA_ARGS
+# echo ***************************************************
```
Comment (chlafreniere) [inline scripts/test-extensions-unit.sh:94]: please undo changes here

**Review by chlafreniere — APPROVED** [review]

**Review by VasuBhog — APPROVED** [review]

**Pull request merged**